src/app/cli/src/commands/system_info_command.rs

```diff
@@ -164,3 +184,58 @@ impl BuildInfo {
         }
     }
 }
+
+#[derive(Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WorkspaceInfo {
+    pub version: u32,
+    pub root_dir: String,
+}
+
+impl WorkspaceInfo {
+    pub fn collect(workspace_svc: Arc<WorkspaceService>) -> Self {
+        if !workspace_svc.is_in_workspace() {
+            return Self {
+                version: 0,
+                root_dir: "".to_string(),
+            };
+        }
+        let root_dir = match workspace_svc.layout() {
+            Some(wl) => wl.root_dir.to_str().map(String::from).unwrap(),
+            None => "".to_string(),
+        };
+        Self {
+            version: workspace_svc.workspace_version().unwrap().unwrap().into(),
+            root_dir,
+        }
+    }
+}
+
+#[derive(Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ContainerRuntimeInfo {
+    pub info: Value,
+}
+
+impl ContainerRuntimeInfo {
+    pub async fn collect(container_runtime: Arc<ContainerRuntime>) -> Self {
+        let container_info_output = match container_runtime.info().output().await {
+            Ok(container_info) => {
+                if container_info.status.success() {
+                    let output = String::from_utf8(container_info.stdout).unwrap();
+                    Value::from_str(output.as_str()).unwrap()
+                } else {
+                    Value::from_str("{\"error\":\"Unable to parse container runtime info result\"}")
+                        .unwrap()
```

Comment:
Instead of parsing a string you could express this as
```rust
serde_json::json!({"error": "blah"})
```
which would construct a `Value` at compile time.